### PR TITLE
Support user-level .dazelrc

### PR DESCRIPTION
**Pull request by scele**
- Support user config file in ~/.dazelrc.  This can be used for adding
  user-specific configuration such as extra mounts.
- Refactor config loading so that it's done in steps, and each step can
  modify the config state of the previous steps (e.g. append to DAZEL_VOLUMES
  array):
     1) Initialize defaults
     2) Read user config file from ~/.dazelrc
     3) Read workspace's .dazelrc
     4) Apply environment overrides
- Automatically add .dazelrc and ~/.dazelrc to FILES_WATCH, so users
  can't forget adding them.
- Expand ~ to user home directory in DAZEL_VOLUMES
- Replace types.Iterable (doesn't work in python3) with collections.Iterable

### CODE REVIEW

**Comment by mirandaconrado** [comment]
Should we change the order such that the user config can override the workspace one? I think the order should be more generic to more specific:
- defaults (global config)
- workspace (team config)
- home (person config)
- env (temp config)

What do you think? This would also be more similar to [bazelrc](https://blog.bazel.build/2015/07/01/Configuration-File.html).

File: dazel.py [hunk 1]
```diff
@@ -133,59 +133,78 @@ def __init__(self, instance_name, image_name, run_command, docker_command, docke
...
+            "DAZEL_TIMEOUT_WATCH_TIME": DEFAULT_TIMEOUT_WATCH_TIME,
+            "DAZEL_TIMEOUT_WATCH_PATH": DEFAULT_TIMEOUT_WATCH_PATH,
+            "DAZEL_IMAGE_TAG": DEFAULT_IMAGE_TAG,
+            "DAZEL_CLEANUP_TIMESCALE": DEFAULT_CLEANUP_TIMESCALE,
+            "DAZEL_BUILD_ARGS": DEFAULT_BUILD_ARGS,
+        }
+        # Allow overriding defaults first in the user config file, then in the
+        # workspace config file, and finally via environment.
+        config.update(cls._config_from_user_file(config))
+        config.update(cls._config_from_workspace_file(config))
```
Comment (mirandaconrado) [inline dazel.py]: Maybe switch 172 and 173?

**Comment by scele** [comment]
I followed the .gitconfic order of precedence (https://git-scm.com/docs/git-config#FILES):
- /etc/gitconfig (system wide)
- ~/.gitconfig (user)
- $GIT_DIR/config (project specific)

If you think a different order makes more sense (and particularly if bazel follows a different order), I'd be totally fine with that! Maybe you should just amend this commit to your liking and then merge?

**Comment by mirandaconrado** [comment]
Oh, interesting! Didn't know git did this way. Yeah, let's merge and maybe change later (should be OK since we're going to add instead of overwriting).

**Pull request merged**